app/controllers/ats/candidates_controller.rb

```diff
@@ -432,24 +432,44 @@ def upload_file
       actor_account: current_account,
       file: candidate_params[:file]
     ).call
-    in Success(file)
+    in Success()
       redirect_to tab_ats_candidate_path(@candidate, :files)
     in Failure[:file_invalid, e]
       render_error e, status: :unprocessable_entity
     end
   end
 
   def upload_cv_file
-    case Candidates::UploadFile.new(
-      candidate: @candidate,
-      actor_account: current_account,
-      file: candidate_params[:file],
-      cv: true
-    ).call
-    in Success(file)
-      redirect_to tab_ats_candidate_path(@candidate, :info)
-    in Failure[:file_invalid, e]
-      render_error e, status: :unprocessable_entity
+    file = candidate_params[:file]
+    # TODO: remove transaction
+    ActiveRecord::Base.transaction do
+      case Candidates::UploadFile.new(
+        candidate: @candidate,
+        actor_account: current_account,
+        file:,
+        cv: true
+      ).call
+      in Success()
+        case Candidates::UpdateFromCV.new(
+          candidate: @candidate,
+          actor_account: current_account,
+          cv_file: file
+        ).call
+        in Success()
+          nil
+        in Failure[:contacts_invalid, data]
+          warning = t("candidates.upload_cv.contacts_invalid", data:)
+        in Failure[:invalid_pdf, data]
+          warning = t("candidates.upload_cv.invalid_pdf", data:)
+        in Failure[:parse_pdf_error, data]
+          warning = t("candidates.upload_cv.parse_pdf_error", data:)
```

Comment:
I want to group these, but idk how to get failure's message (the only way I see is to pass it as an additional value)

Reply:
actually, I just followed the usual workflow to handle every possible error. But I'm fine with sending it to sentry

Reply:
yeah, we handle, but display nothing to user, just for us in sentry